messaging/src/main/java/org/axonframework/eventhandling/pooled/Coordinator.java

```diff
@@ -839,6 +841,26 @@ private CompletableFuture<Void> abortWorkPackages(Exception cause) {
                                .thenRun(workPackages::clear);
         }
 
+
+        /**
+         * Compares the maximum number of segments that can be claimed
+         * by a node for a fair distribution of the segments among available processor nodes.
+         * Releases extra segments claimed by this event processor instance to be available for claim by other processors
+         * @return true if segments claimed by the Node > MaxSegmentProvider.getMaxSegments
```

Comment:
I think, more in line with the methodname, something like `@return true if any segments were released` would be better. Also, I think it's a bit too precise for your use case. Whatever the reason the max segment provider is lower than the current claimed segments, it will release segments to respect the maximum.

Reply:
Rephrased it as suggested.